src/renderers/shared/fiber/ReactChildFiber.js

```diff
@@ -62,6 +64,31 @@ const {
   Deletion,
 } = ReactTypeOfSideEffect;
 
+function transferRef(current: ?Fiber, workInProgress: Fiber, element: ReactElement<any>) {
+  if (typeof element.ref === 'string') {
+    if (element._owner) {
+      const ownerFiber : ?Fiber = (element._owner : any);
+      if (ownerFiber && ownerFiber.tag === ClassComponent) {
+        const stringRef = element.ref;
+        // Check if previous string ref matches new string ref
+        if (current && current.ref && current.ref._stringRef === stringRef) {
+          workInProgress.ref = current.ref;
+          return;
+        }
+        const inst = ownerFiber.stateNode;
+        const ref = function(value) {
```

Comment:
This function won't get a name. We should make any functions calling user code named for better stack traces.